library/Pipeline/src/Juvix/Pipeline.hs

```diff
@@ -19,26 +20,70 @@ import qualified Juvix.Core.Base as Core
 import qualified Juvix.Core.Base.TransformExt as TransformExt
 import qualified Juvix.Core.Base.TransformExt.OnlyExts as OnlyExts
 import qualified Juvix.Core.Erased.Ann as ErasedAnn
+import qualified Juvix.Core.HR.Types as HR
 import qualified Juvix.Core.IR as IR
 import qualified Juvix.Core.IR.Typechecker.Types as TypeChecker
 import Juvix.Core.Parameterisation
   ( CanApply (ApplyErrorExtra, Arg),
     TypedPrim,
   )
 import qualified Juvix.Core.Parameterisation as Param
+import qualified Juvix.Core.Translate as Translate
 import qualified Juvix.Core.Types as Core
+import qualified Juvix.Frontend as Frontend
+import qualified Juvix.Frontend.Types as Types
 import Juvix.Library
 import qualified Juvix.Library.Feedback as Feedback
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import Juvix.Library.Parser (ParserError)
 import Juvix.Pipeline.Compile
-import Juvix.Pipeline.Internal
-import qualified Juvix.Pipeline.Internal as Pipeline
+import qualified Juvix.Pipeline.Core as Core
+import qualified Juvix.Pipeline.Frontend as Frontend
 import Juvix.Pipeline.Types
 import qualified Juvix.Sexp as Sexp
-import Juvix.ToCore.FromFrontend as FF (CoreDefs (..))
+import qualified Juvix.ToCore.FromFrontend as FF
 import qualified System.IO.Temp as Temp
 import qualified Text.Megaparsec as P
+import Text.Pretty.Simple (pShowNoColor)
 import qualified Text.PrettyPrint.Leijen.Text as Pretty
 
+-- TODO: Change error type to Error
+type Pipeline = Feedback.FeedbackT [] [Char] IO
```

Comment:
As discussed, we might want to change `[Char]` not to `Error`, but to a message type. Something along the lines of:
```haskell
data Msg
  = Info String
  | Warning String
  | Error String
```